bonfire/elastic_logging.py

```diff
@@ -0,0 +1,82 @@
+from datetime import datetime as dt
+import logging
+import json
+import requests
+import sys
+import uuid
+from concurrent.futures import ThreadPoolExecutor
+
+import bonfire.config as conf
+
+
+log = logging.getLogger(__name__)
+
+
+class ElasticLogger():
+    def __init__(self):
+        self.es_telemetry = logging.getLogger("elasicsearch")
+        metadata = {
+            "uuid": str(uuid.uuid4()),
+            "start_time": dt.now().isoformat(),
+            "bot": conf.BONFIRE_BOT,
+            "command": self._mask_parameter_values(sys.argv[1:])
+        }
+
+        es_handler = next((h for h in self.es_telemetry.handlers
+                           if type(h) is AsyncElasticsearchHandler), None)
+        if es_handler:
+            log.warning("AsyncElasticsearchHandler already configured for current logger")
+
+        self.es_handler = AsyncElasticsearchHandler(conf.ELASTICSEARCH_HOST, metadata)
+        self.es_telemetry.addHandler(self.es_handler)
+
+    def _mask_parameter_values(self, cli_args):
+        masked_list = []
+
+        is_parameter = False
+        for arg in cli_args:
+            if is_parameter:
+                masked_arg = f"{arg.split('=')[0]}=*******"
+                masked_list.append(masked_arg)
+                is_parameter = False
+            else:
+                masked_list.append(arg)
+                is_parameter = arg == '-p' or arg == '--set-parameter'
+
+        return masked_list
+
+    def send_telemetry(self, log_message, success=True):
+        self.es_handler.set_success_status(success)
+
+        self.es_telemetry.info(log_message)
+
+
+class AsyncElasticsearchHandler(logging.Handler):
+    def __init__(self, es_url, metadata):
+        super().__init__()
+        self.es_url = es_url
+        self.metadata = metadata
+        self.executor = ThreadPoolExecutor(max_workers=10)
+
+    def emit(self, record):
+        self.metadata["time_of_logging"] = dt.now().isoformat()
+        self.metadata["duration"] = str(dt.now() - dt.fromisoformat(self.metadata["start_time"]))
```

Comment:
ok I did that. The only caveat is that I need to format the start_time before I pass the metadata to the request. I add a small piece that makes a copy of the metadata that's formatted.
I didn't want to modify the original metadata just in case there would be multiple logs each run.